### PR TITLE
feat(engine): Add BulkAttributeRenamer processor

### PR DESCRIPTION
**Pull request by soma00333**
# Overview
Add BulkAttributeRenamer processor.
Related Issue: https://github.com/reearth/reearth-flow/issues/574
### Add BulkAttributeRenamer
Create bulk_renamer.rs
Schema is below

＊Parameter
- rename type
  - all or selected attribute
- selected attribute
  - array
- rename action
  - add_prefix, add_suffix, remove_prefix, remove_suffix, string_replace
- text to find (only string_replace)
  - string
- rename value
  - string
  
＊Port
- Input port
  - default
- Output port
  - default
  
＊Category
- attribute

### Add example workflow files and integration tests
Add below workflows and integration tests
- all_add_prefix
- all_add_suffix
- all_remove_prefix_error
- all_remove_prefix
- all_remove_suffix_error
- all_remove_suffix
- all_string_replace_error
- all_string_replace
- delete_attribute
- selected_add_prefix
- selected_add_suffix
- selected_remove_prefix_error
- selected_remove_prefix
- selected_remove_suffix_error
- selected_remove_suffix
- selected_string_replace_error
- selected_string_replace

## What I haven't done
Unit test

## How I tested
Checked the output of all workflows I added
Checked with `make cargo test` 

## Which point I want you to review particularly
Currently, the methods remove_prefix, remove_suffix, and string_replace return errors if an attribute name does not match the expected prefix, suffix, or regex pattern.
Should we modify these methods to skip non-matching attributes? 

## Summary by CodeRabbit

- **New Features**
	- Introduced the `BulkAttributeRenamer` processor for flexible attribute renaming, including options to add/remove prefixes/suffixes or replace text.
	- Added multiple workflow examples demonstrating various renaming operations (e.g., adding/removing prefixes/suffixes, string replacements).
	- Implemented a new module for bulk renaming, expanding the existing functionality.
	- Added new test cases to validate various bulk renaming scenarios, ensuring comprehensive coverage of functionalities.

- **Bug Fixes**
	- Enhanced error handling for bulk renaming operations.

- **Documentation**
	- New YAML configuration files added to showcase different bulk renaming scenarios and their configurations.
	- Added unit tests to validate bulk renaming functionalities.

### CODE REVIEW

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+impl Processor for BulkAttributeRenamer {
+    fn process(
+        &mut self,
+        ctx: ExecutorContext,
+        fw: &mut dyn ProcessorChannelForwarder,
+    ) -> Result<(), BoxedError> {
+        let mut feature = ctx.feature.clone();
+
+        let attributes_to_rename = self.select_attributes(&feature)?;
+        let attributes_to_remove = self.rename_attributes(&mut feature, attributes_to_rename)?;
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:114]: Select attributes and take the required action

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+        for attr in attributes_to_remove {
+            feature.attributes.remove(&attr);
+        }
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:118]: Remove all old attributes

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        Ok(())
+    }
+
+    fn name(&self) -> &str {
+        "BulkAttributeRenamer"
+    }
+}
+
+impl BulkAttributeRenamer {
+    fn select_attributes(&self, feature: &Feature) -> Result<Vec<Attribute>, BoxedError> {
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:138]: Logic for attribute selection

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                    Err(AttributeProcessorError::BulkRenamer(
+                        "No attributes selected for renaming".to_string(),
+                    )
+                    .into())
+                }
+            }
+        }
+    }
+
+    fn rename_attributes(
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:154]: Logic for renaming attributes

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+
+        for attr in attributes {
+            if let Some(value) = feature.attributes.get(&attr) {
+                let new_name = self.get_new_name(&attr.inner())?;
+                if new_name.is_empty() {
+                    feature.attributes.remove(&attr);
+                } else {
+                    let new_attr = Attribute::new(new_name);
+                    feature.attributes.insert(new_attr, value.clone());
+                    attributes_to_remove.push(attr.clone());
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:169]: Push to remove old attributes

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+                }
+            }
+        }
+        Ok(attributes_to_remove)
+    }
+
+    fn get_new_name(&self, attr_name: &str) -> Result<String, BoxedError> {
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:176]: A helper function to return new attribute names

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        match self.params.rename_action {
+            RenameAction::AddPrefix => Ok(format!("{}{}", self.params.rename_value, attr_name)),
+            RenameAction::AddSuffix => Ok(format!("{}{}", attr_name, self.params.rename_value)),
+            RenameAction::RemovePrefix => self.remove_prefix(attr_name),
+            RenameAction::RemoveSuffix => self.remove_suffix(attr_name),
+            RenameAction::StringReplace => self.string_replace(attr_name),
+        }
+    }
+
+    fn remove_prefix(&self, attr_name: &str) -> Result<String, BoxedError> {
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:186]: Prefix deletion process

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+        } else {
+            Err(AttributeProcessorError::BulkRenamer(format!(
+                "Attribute '{}' does not start with prefix '{}'",
+                attr_name, self.params.rename_value
+            ))
+            .into())
+        }
+    }
+
+    fn remove_suffix(&self, attr_name: &str) -> Result<String, BoxedError> {
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:201]: Suffix deletion process

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+        } else {
+            Err(AttributeProcessorError::BulkRenamer(format!(
+                "Attribute '{}' does not end with suffix '{}'",
+                attr_name, self.params.rename_value
+            ))
+            .into())
+        }
+    }
+
+    fn string_replace(&self, attr_name: &str) -> Result<String, BoxedError> {
```
Comment (soma00333) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:216]: String replacement process
We can use regular expressions here

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+                }
+            }
+        }
+    }
+
+    fn rename_attributes(
+        &self,
+        feature: &mut Feature,
+        attributes: Vec<Attribute>,
+    ) -> Result<Vec<Attribute>, BoxedError> {
```
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:158]:
```suggestion
    ) -> super::errors::Result<Vec<Attribute>> {
```

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 3 above)
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:138]:
```suggestion
    fn select_attributes(&self, feature: &Feature) -> super::errors::Result<Vec<Attribute>> {
```

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 6 above)
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:176]:
```suggestion
    fn get_new_name(&self, attr_name: &str) -> super::errors::Result<String> {
```

Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:176]:
```suggestion
    fn get_new_name(&self, attr_name: &str) -> super::errors::Result<String> {
```

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 7 above)
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:186]:
```suggestion
    fn remove_prefix(&self, attr_name: &str) -> super::errors::Result<String> {
```

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 8 above)
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:201]:
```suggestion
    fn remove_suffix(&self, attr_name: &str) -> super::errors::Result<String> {
```

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+            Err(AttributeProcessorError::BulkRenamer(format!(
+                "Attribute '{}' does not end with suffix '{}'",
+                attr_name, self.params.rename_value
+            ))
+            .into())
+        }
+    }
+
+    fn string_replace(&self, attr_name: &str) -> Result<String, BoxedError> {
+        if let Some(ref find) = self.params.text_to_find {
```
Comment (miseyu) [inline engine/runtime/action-processor/src/attribute/bulk_renamer.rs:217]: must
Regular expressions should be compiled in advance using the build method.

File: engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml [hunk 12]
```diff
@@ -0,0 +1,53 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/reearth/reearth-flow/main/engine/schema/workflow.json
+id: 00caad2a-9f7d-4189-b479-153fa9ea36dc
```
Comment (miseyu) [inline engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml:2]: @soma00333 
I'll try to make it easier to write test cases. We will try to send you an example later.

**Comment by miseyu** [comment]
@soma00333 
I have reviewed it. I have made some comments, please check them out. 🙇

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 10 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 3 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 6 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 7 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 8 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/79c0e5216ee2a36bdfd4ec3dda3a4f462e10ab83

File: engine/runtime/action-processor/src/attribute/bulk_renamer.rs (hunk 11 above)
Comment (soma00333) [reply engine/runtime/action-processor/src/attribute/bulk_renamer.rs]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/fef6798bec637fdb35e24df76a9072257727df2a

File: engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml (hunk 12 above)
Comment (miseyu) [reply engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml]: @soma00333 
I have moved it from examples to tests.
I have written three tests for now, but I would appreciate it if you could also prepare test results (expect.json) for other test cases.

https://github.com/reearth/reearth-flow/pull/583/commits/b7ba4ce1e2791444ea1ccce4be1f1648dff1ecef

Comment (soma00333) [reply engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml]: @miseyu 
Thank you. I’ll take care of the remaining tests👍

Comment (soma00333) [reply engine/runtime/examples/plateau/testdata/workflow/example_bulk_renamer/all_add_prefix.yml]: Fixed here https://github.com/reearth/reearth-flow/pull/583/commits/db4e709104ded3ca17bfc56bb7d11c4d3beb636e

**Review by miseyu — APPROVED** [review]
Thank you so much!
LGTM

**Pull request merged**